lib/jsdom/living/window/SessionHistory.js

```diff
@@ -0,0 +1,60 @@
+"use strict";
+const traverseHistory = require("./navigation.js").traverseHistory;
+
+// https://html.spec.whatwg.org/#session-history
+class SessionHistory {
+  constructor(initialEntry, window) {
+    this._window = window;
+    this._historyTraversalQueue = new Set();
+    this.entries = [initialEntry];
+    this.currentIndex = 0;
```

Comment:
Do you think it'd be a good idea to encapsulate entries/currentIndex entirely? It seems a bit strange that sometimes you go through sessionHistory's properties (e.g. length, currentEntry, removeALlEntriesAfterCurrentEntry) and sometimes you directly manipulate the entries array.

Also, below you use an underscore in currentIndex. It seems bad that we don't have any tests that would catch that :-/.